src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -537,22 +543,36 @@ private void repartitionEventType(
     }
 
     private void updateNumberOfPartitions(final EventType original, final EventType eventType, final int partitions)
-            throws InternalNakadiException {
+            throws InternalNakadiException, NakadiRuntimeException {
         final NakadiBaseException exception = transactionTemplate.execute(action -> {
             try {
                 eventTypeRepository.update(eventType);
                 timelineService.updateTimeLineForRepartition(eventType, partitions);
                 return null;
-            } catch (CannotAddPartitionToTopicException | InternalNakadiException | TopicConfigException e) {
+            } catch (NakadiBaseException e) {
                 return e;
             }
         });
 
         if (exception != null) {
             throw new InternalNakadiException("Cannot repartition Event type " + original.getName(), exception);
         }
+
+        updateSubscriptionsForRepartitioning(eventType, partitions);
     }
 
+    public void updateSubscriptionsForRepartitioning(final EventType eventType, final int partitions)
+            throws NakadiBaseException {
+        final List<Subscription> subscriptions = subscriptionRepository.listSubscriptions(
+                ImmutableSet.of(eventType.getName()), Optional.empty(), 0, 1000);
+        for (final Subscription subscription : subscriptions) {
+            final ZkSubscriptionClient zkClient = subscriptionClientFactory.createClient(subscription,
+                    LogPathBuilder.build(subscription.getId(), "repartition"));
+            zkClient.repartitionTopology(eventType.getName(), partitions);
+            zkClient.closeSubscriptionStreams(
+                    () -> LOG.info("subscription streams were closed, after repartitioning"), 60000);
```

Comment:
Can we use a constant (and also in line 567?)

Reply:
done